File: Source/Aladdin.ASN1.ISO.Java/src/aladdin/asn1/iso/pkcs/pkcs12/CRLBag.java
package aladdin.asn1.iso.pkcs.pkcs12;
import aladdin.asn1.*; 
import java.io.*;

//	CRLBag ::= SEQUENCE {
//		crlId			         OBJECT IDENTIFIER,
//		crltValue	[0] EXPLICIT ANY DEFINED BY crlId
//	}

public final class CRLBag extends Sequence<IEncodable>
{
	// информация о структуре
	private static final ObjectInfo[] info = new ObjectInfo[] { 

		new ObjectInfo(new ObjectCreator(ObjectIdentifier.class).factory(), Cast.N,	Tag.ANY			), 
		new ObjectInfo(    ImplicitCreator					    .factory  , Cast.E,	Tag.context(0)	), 
	}; 
	// конструктор при раскодировании
	public CRLBag(IEncodable encodable) throws IOException { super(encodable, info); }

	// конструктор при закодировании
	public CRLBag(ObjectIdentifier crlId, IEncodable crltValue) 
	{
		super(info, crlId, crltValue); 
	}
	public final ObjectIdentifier	crlId	 () { return (ObjectIdentifier)get(0); } 
	public final IEncodable         crltValue() { return	 			   get(1); }
}

File: Source/Aladdin.ASN1.ISO/ASN1/ISO/PKCS/PKCS12/CRLBag.cs
using System;
using System.Runtime.Serialization;

//	CRLBag ::= SEQUENCE {
//		crlId			         OBJECT IDENTIFIER,
//		crltValue	[0] EXPLICIT ANY DEFINED BY crlId
//	}

namespace Aladdin.ASN1.ISO.PKCS.PKCS12
{
	[Serializable]
	public class CRLBag : Sequence
	{
		// информация о структуре
		private static readonly ObjectInfo[] info = new ObjectInfo[] { 

			new ObjectInfo(new ObjectCreator<ObjectIdentifier>().Factory(), Cast.N,	Tag.Any			), 
			new ObjectInfo(    ImplicitCreator				    .Factory  , Cast.E,	Tag.Context(0)	), 
		}; 
		// конструктор при сериализации
        protected CRLBag(SerializationInfo info, StreamingContext context) : base(info, context) {}

		// конструктор при раскодировании
		public CRLBag(IEncodable encodable) : base(encodable, info) {}

		// конструктор при закодировании
		public CRLBag(ObjectIdentifier crlId, IEncodable crltValue) : 
			base(info, crlId, crltValue) {}

		public ObjectIdentifier	CrlId		{ get { return (ObjectIdentifier)this[0]; } } 
		public IEncodable		CrltValue	{ get { return					 this[1]; } }
	}
}
